### PR TITLE
improvement(sdcm.argus_test_run): Azure resources config support

### PR DESCRIPTION
**Pull request by k0machi**
[Trello](https://trello.com/c/Filaar8E/4958-add-azure-support-in-argus)

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [x] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [x] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [x] All new and existing unit tests passed (CI)
- [x] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by k0machi** [comment]
Running a [job](https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/alexey/job/longevity-10gb-3h-azure-test/1/) to verify.

**Review by soyacz — APPROVED** [review]
I cannot say much about it without deeper understanding Argus integration code. 
Anyway despite one small thing: LGTM :)

File: sdcm/argus_test_run.py [hunk 1]
```diff
@@ -86,6 +86,26 @@ def _prepare_gce_resource_setup(sct_config: SCTConfiguration):
     return cloud_setup
 
 
+def _prepare_azure_resource_setup(sct_config: SCTConfiguration):
```
[the remark this answers is not in the document]
Comment (k0machi) [reply sdcm/argus_test_run.py:89]: Fixed

File: sdcm/argus_test_run.py [hunk 2]
```diff
@@ -86,6 +86,26 @@ def _prepare_gce_resource_setup(sct_config: SCTConfiguration):
...
+                                   post_behaviour=sct_config.get("post_behavior_db_nodes"))
+    loader_node_setup = CloudNodesInfo(image_id=sct_config.get("azure_image_loader"),
+                                       instance_type=sct_config.get("azure_instance_type_loader"),
+                                       node_amount=n_loaders,
+                                       post_behaviour=sct_config.get("post_behavior_loader_nodes"))
+    monitor_node_setup = CloudNodesInfo(image_id=sct_config.get("azure_image_monitor"),
+                                        instance_type=sct_config.get("azure_instance_type_monitor"),
+                                        node_amount=sct_config.get("n_monitor_nodes"),
+                                        post_behaviour=sct_config.get("post_behavior_monitor_nodes"))
+    cloud_setup = AWSSetupDetails(db_node=db_node_setup, loader_node=loader_node_setup,
```
[the remark this answers is not in the document]
Comment (k0machi) [reply sdcm/argus_test_run.py:103]: Good catch, it should be BaseCloudSetupDetails here

**Review by soyacz — APPROVED** [review]

**Review by bentsi — APPROVED** [review]

**Pull request merged**